adalflow/adalflow/components/model_client/bedrock_client.py

```diff
@@ -0,0 +1,154 @@
+
+    def init_sync_client(self):
+        """
+        There is no need to pass both profile and secret key and access key. Path one of them.
+        if the compute power assume a role that have access to bedrock, no need to pass anything.
+        """
+        aws_profile_name = self._aws_profile_name or os.getenv("AWS_PROFILE_NAME")
+        aws_region_name = self._aws_region_name or os.getenv("AWS_REGION_NAME")
+        aws_access_key_id = self._aws_access_key_id or os.getenv("AWS_ACCESS_KEY_ID")
+        aws_secret_access_key = self._aws_secret_access_key or os.getenv("AWS_SECRET_ACCESS_KEY")
+        aws_session_token = self._aws_session_token or os.getenv("AWS_SESSION_TOKEN")
+
+        config = None
+        if self._aws_connection_timeout or self._aws_read_timeout:
+            config = Config(
+                connect_timeout=self._aws_connection_timeout,  # Connection timeout in seconds
+                read_timeout=self._aws_read_timeout  # Read timeout in seconds
+            )
+
+        session = boto3.Session(
+            profile_name=aws_profile_name,
+            region_name=aws_region_name,
+            aws_access_key_id=aws_access_key_id,
+            aws_secret_access_key=aws_secret_access_key,
+            aws_session_token=aws_session_token,
+        )
+        bedrock_runtime = session.client(service_name="bedrock-runtime", config=config)
+        return bedrock_runtime
+
+    def init_async_client(self):
+        raise NotImplementedError("Async call not implemented yet.")
+
+    def parse_chat_completion(self, completion):
+        log.debug(f"completion: {completion}")
+        try:
+            data = completion['output']['message']['content'][0]['text']
+            usage = self.track_completion_usage(completion)
+            return GeneratorOutput(data=None, usage=usage, raw_response=data)
+        except Exception as e:
+            log.error(f"Error parsing completion: {e}")
+            return GeneratorOutput(
+                data=None, error=str(e), raw_response=str(completion)
+            )
+
+    def track_completion_usage(self, completion: Dict) -> CompletionUsage:
+        r"""Track the completion usage."""
+        usage = completion['usage']
+        return CompletionUsage(
+            completion_tokens=usage['outputTokens'],
+            prompt_tokens=usage['inputTokens'],
+            total_tokens=usage['totalTokens']
+        )
+
+    def convert_inputs_to_api_kwargs(
+            self,
+            input: Optional[Any] = None,
+            model_kwargs: Dict = {},
+            model_type: ModelType = ModelType.UNDEFINED
+    ):
+        """
+        check the converse api doc here:
+        https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/bedrock-runtime/client/converse.html
+        """
+        api_kwargs = model_kwargs.copy()
+        if model_type == ModelType.LLM:
+            api_kwargs["messages"] = [
+                {"role": "user", "content": [{"text": input}]},
```

Comment:
interesting, so all their converse api take a list as input? Why not use "system", and is this because they have multimodal, there can accept "image" as input?

Reply:
@liyin2015 
AWS recommends it for bedrock using Converse operation over InvokeModel when supported, since it unifies the inference request across Amazon Bedrock models and simplifies the management of multi-turn conversations. They mention "If a model has unique inference parameters, you can also pass those unique parameters to the model. Amazon Bedrock doesn't store any text, images, or documents that you provide as content. The data is only used to generate the response." So I assume yes they can accept image as input.

Source: 
- https://docs.aws.amazon.com/bedrock/latest/userguide/getting-started-api-ex-python.html
-https://docs.aws.amazon.com/bedrock/latest/APIReference/API_runtime_Converse.html#API_runtime_Converse_RequestParameters